System/Random.hs

```diff
@@ -108,45 +108,52 @@
 --     some background on splittable pseudo-random generators.
 --
 -- [Monadic pseudo-random number generators] These generators mutate their own
---     state as they produce random values. They generally live in 'ST' or 'IO'
---     or some transformer that implements 'PrimMonad'. 'MonadRandom' defines
---     the interface for monadic pseudo-random number generators.
+--     state as they produce pseudo-random values. They generally live in 'ST'
+--     or 'IO' or some transformer that implements @PrimMonad@. 'MonadRandom'
+--     defines the interface for monadic pseudo-random number generators.
 --
 -- Pure pseudo-random number generators can be used in monadic code via the
--- adapters 'PureGen', 'PrimGen' and 'MutGen'.
+-- adapters 'PureGen', 'AtomicGen', 'IOGen' and 'STGen'.
 --
 -- *   'PureGen' can be used in any state monad. With strict 'StateT' there is
---     even no performance overhead when compared to the 'RandomGen' directly.
---     But it is not safe to use it in presence of exceptions and resource
---     allocation that requires cleanup.
+--     no performance overhead compared to using the 'RandomGen' instance
+--     directly. 'PureGen' is /not/ safe to use in the presence of exceptions
+--     and concurrency.
+--
+-- *   'AtomicGen' is safe in the presence of exceptions and concurrency since
+--     it performs all actions atomically.
 --
--- *   'PrimGen' can be used in any 'PrimMonad' if the pseudo-random number
---     generator is also an instance of 'Prim'. It will perform much faster
---     than 'MutGen', but it can't be used in a concurrent setting.
+-- *   'IOGen' is a wrapper around an 'IORef' that holds a pure generator.
+--     'IOGen' is /not/ safe to use in the presence of exceptions and
+--     concurrency.
 --
--- *   'MutGen' can be used in any 'PrimMonad' (this includes 'ST' and 'IO') and
---     can safely be shared between threads.
+-- *   'STGen' is a wrapper around an 'STRef' that holds a pure generator.
+--     'STGen' is /not/ safe to use in the presence of exceptions and
+--     concurrency.
 --
 -- When to use which?
 --
 -- *   Use 'PureGen' if your computation does not have side effects and results
 --     in a pure function.
 --
--- *   Use 'PrimGen' if the pseudo-random number generator implements 'Prim'
---     class and random value generation is intermixed with 'IO' or 'ST'.
+-- *   Use 'AtomicGen' if the pseudo-random number generator must be shared
+--     between threads or if exception safety is a requirement.
```

Comment:
"or if exception safety is a requirement." This can be removed.

Reply:
:+1: